src/signing.rs

```diff
@@ -644,11 +640,28 @@ impl<'d> Deserialize<'d> for SigningKey {
     where
         D: Deserializer<'d>,
     {
-        let bytes = <SerdeByteBuf>::deserialize(deserializer)?;
-        Self::try_from(bytes.as_ref()).map_err(SerdeError::custom)
+        struct SigningKeyVisitor;
+
+        impl <'de> serde::de::Visitor<'de> for SigningKeyVisitor {
+            type Value = SigningKey;
+
+            fn expecting(&self, formatter: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
+                write!(
+                    formatter,
+                    concat!("An ed25519 signing (private) key")
+                )
+            }
+
+            fn visit_borrowed_bytes<E: serde::de::Error>(self, bytes: &'de [u8]) -> Result<Self::Value, E> {
+                SigningKey::try_from(bytes.as_ref()).map_err(E::custom)
+            }
```

Comment:
Won't this fail for formats that can't borrow from their inputs, such as JSON and TOML?

It might be better to use `visit_seq`.

Also since this is a private key, it might make sense to use `serdect` which uses constant-time hex serializers for human readable formats: https://github.com/RustCrypto/formats/tree/master/serdect

Reply:
> Won't this fail for formats that can't borrow from their inputs, such as JSON and TOML?

oh interesting, TIL they can't borrow inputs / `visit_seq` makes sense there!

> since this is a private key, it might make sense to use serdect which uses constant-time hex serializers for human readable formats

human readable is always nice, but, that seems like it would be a change in behaviour / incompatible with binary serialisation?

Reply:
> human readable is always nice, but, that seems like it would be a change in behaviour / incompatible with binary serialisation?

It uses binary serialization with non-human-readable formats

Reply:
okay looking at `serdect` again, it still seems that adding this would be a breaking change to the way values are serialised compared to previous versions, which might be worthwhile (love a good human readable encoding) but, is probably outside the scope of this PR?

Reply:
Okay, for the sake of fixing the `serde` + `no_std` serde build, we can start with this.

However, it is a breaking release, and it's really bad if people are depending on their private keys being serialized as `[12, 34, 56, ...]`.

Reply:
> However, it is a breaking release, and it's really bad if people are depending on their private keys being serialized as [12, 34, 56, ...].

yeah fully agreed on that front! and it'd be a nice usability improvement to not need to override the serde encode/decode manually for those of us who noticed / worry about these things.